### PR TITLE
Adds Docker files to hyku-api

### PR DESCRIPTION
**Pull request by BertZZ**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Base stage for building final images
+FROM phusion/passenger-ruby27:2.0.0 as BASE_IMAGE
```
Comment (prdanelli) [inline Dockerfile:2]: I think Tom was having issues with the capitalized BASE_IMAGE/DEVELOPMENT_IMAGE names. Could you lowercase them here and we can make the same changes on Hyku Addons :)

(Dont forget to change the image name in the docker-compose file too)

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+services:
+  app: &app
+    build:
+      context: .
+      dockerfile: Dockerfile
+      target: DEVELOPMENT_IMAGE
+    image: hyku-api/development_image:latest
+    environment: &web_environment
+      HOST: "0.0.0.0" # Docker cannot forward to a containers localhost, so we need to use this IP instead
```
Comment (prdanelli) [inline docker-compose.yml:25]: Can you check if any of these settings can be removed as they're not required for the API?

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+      REDIS_NAMESPACE_QUIET: "true" # Turn off Gem deprecations warnings
+    volumes:
+      - app:/home/app/spec/internal_test_hyku/tmp/uploads
+      - assets:/home/app/spec/internal_test_hyku/public/assets
+      - cache:/home/app/spec/internal_test_hyku/tmp/cache
+      - .:/home/app
+    networks:
+      internal:
+
+  web:
```
Comment (prdanelli) [inline docker-compose.yml:64]: Can any of the services be removed as they're not used by the API?

Comment (mankind) [reply docker-compose.yml:53]: May be looking at the hyrax-orcid docker-compose file will help you  see services you might not need. I looked at it for the docker-compose in hyrax-autopopulation, which you can still check.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	vim
+
+RUN apt clean \
+	&& rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+# If changes are made to fits version or location, amend `LD_LIBRARY_PATH` in docker-compose.yml accordingly.
+RUN mkdir -p /opt/fits && \
+    curl -fSL -o /opt/fits/fits-latest.zip https://projects.iq.harvard.edu/files/fits/files/fits-1.3.0.zip && \
+    cd /opt/fits && unzip fits-latest.zip && \
+		chmod +X /opt/fits/fits.sh
```
Comment (prdanelli) [inline Dockerfile:31]: Can you fix this indentation and then we're good to go :)

**Review by prdanelli — COMMENTED** [review]
Just a little tweak and we're good to go :)

File: Dockerfile (hunk 4 above)
Comment (prdanelli) [reply Dockerfile:31]: It'll be a spaces/tabs issue i think.

**Pull request merged**